Tools/autotest/quadplane.py

```diff
@@ -2198,9 +2197,107 @@ def DoRepositionTerrain(self):
+            loc1,
+            accuracy=10,
+            timeout=600,
+            height_accuracy=10,
+        )
+        self.delay_sim_time(10)
+        self.wait_altitude(
+            loc1.alt-5,
+            loc1.alt+5,
+            minimum_duration=10,
+            timeout=30,
+            relative=False,
+            altitude_source="TERRAIN_REPORT.current_height"
+        )
+        self.progress(f"terrain_height range {terrain_height_min:.1f} {terrain_height_max:.1f}")
+        if terrain_height_min < min(start_alt,loc1.alt) - 10 or terrain_height_max > max(start_alt,loc1.alt)+10:
+            raise NotAchievedException(f"terrain range breach {start_alt:.1f} {terrain_height_min:.1f} {terrain_height_max:.1f}")
+
+        start_alt = self.mav.recv_match(type='TERRAIN_REPORT', blocking=True, timeout=5).current_height
+        terrain_height_min = start_alt
+        terrain_height_max = start_alt
+
+        self.progress(f"Flying to loc2 at {loc2.alt:.1f} from {start_alt:.1f}")
+        self.send_do_reposition(loc2, frame=mavutil.mavlink.MAV_FRAME_GLOBAL_TERRAIN_ALT)
+
+        self.wait_location(
+            loc2,
+            accuracy=10,
+            timeout=600,
+            height_accuracy=10,
+        )
+        self.delay_sim_time(10)
+        self.wait_altitude(
+            loc2.alt-5,
+            loc2.alt+5,
+            minimum_duration=10,
+            timeout=30,
+            relative=False,
+            altitude_source="TERRAIN_REPORT.current_height"
+        )
+        self.progress(f"terrain_height range {terrain_height_min:.1f} {terrain_height_max:.1f}")
+        if terrain_height_min < min(start_alt,loc2.alt) - 10 or terrain_height_max > max(start_alt,loc2.alt)+10:
+            raise NotAchievedException(f"terrain range breach {start_alt:.1f} {terrain_height_min:.1f} {terrain_height_max:.1f}")
+
+        start_alt = self.mav.recv_match(type='TERRAIN_REPORT', blocking=True, timeout=5).current_height
+        terrain_height_min = start_alt
+        terrain_height_max = start_alt
+
+        self.progress(f"Flying back to loc1 at {loc1.alt:.1f} from {start_alt:.1f}")
+        self.send_do_reposition(loc1, frame=mavutil.mavlink.MAV_FRAME_GLOBAL_TERRAIN_ALT)
+
+        self.wait_location(
+            loc1,
+            accuracy=10,
+            timeout=600,
+            height_accuracy=10,
+        )
+        self.delay_sim_time(10)
+        self.wait_altitude(
+            loc1.alt-5,
+            loc1.alt+5,
+            minimum_duration=10,
+            timeout=30,
+            relative=False,
+            altitude_source="TERRAIN_REPORT.current_height"
+        )
+        self.progress(f"terrain_height range {terrain_height_min:.1f} {terrain_height_max:.1f}")
+        if terrain_height_min < min(start_alt,loc1.alt) - 10 or terrain_height_max > max(start_alt,loc1.alt)+10:
+            raise NotAchievedException(f"terrain range breach {start_alt:.1f} {terrain_height_min:.1f} {terrain_height_max:.1f}")
```

Comment:
Is there some reason you don't want to die instantly when you breach your boundaries?

Reply:
yes, I want to see the final height